client/notifications/notifications-popup.tsx

```diff
@@ -0,0 +1,171 @@
+import { Set } from 'immutable'
+import { lighten } from 'polished'
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import ReactDOM from 'react-dom'
+import {
+  animated,
+  AnimationResult,
+  Controller,
+  useTransition,
+  UseTransitionProps,
+} from 'react-spring'
+import styled from 'styled-components'
+import { assertUnreachable } from '../../common/assert-unreachable'
+import { NotificationType } from '../../common/notifications'
+import { EmailVerificationNotificationUi } from '../auth/email-verification-notification-ui'
+import { useExternalElementRef } from '../dom/use-external-element-ref'
+import CheckIcon from '../icons/material/check-24px.svg'
+import { shadow6dp } from '../material/shadows'
+import { defaultSpring } from '../material/springs'
+import { zIndexMenu } from '../material/zindex'
+import { useAppDispatch, useAppSelector } from '../redux-hooks'
+import { usePrevious } from '../state-hooks'
+import { blue900, grey800 } from '../styles/colors'
+import { markLocalNotificationsRead, markNotificationsRead } from './action-creators'
+import { NotificationRecord } from './notification-reducer'
+
+const POPOVER_DURATION = 10000
+
+const PopupsContainer = styled.div`
+  position: fixed;
+  right: 32px;
+  bottom: 32px;
+  width: 368px;
+  max-height: calc(100% - 112px);
+  z-index: ${zIndexMenu};
+
+  display: flex;
+  flex-direction: column;
+  align-items: flex-end;
+
+  overflow-y: auto;
+  &::-webkit-scrollbar {
+    width: 0px;
+  }
+`
+
+const Popup = styled(animated.div)`
+  display: flex;
+  flex-direction: row;
+  align-items: flex-start;
+  background-color: ${grey800};
+  ${shadow6dp};
+  border-radius: 2px;
+
+  &:not(:first-child) {
+    margin-top: 16px;
+  }
+`
+
+const MarkAsReadContainer = styled.div`
+  flex-shrink: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 48px;
+  height: 100%;
+  cursor: pointer;
+  background-color: ${blue900};
+
+  &:hover {
+    background-color: ${lighten(0.08, blue900)};
```

Comment:
Hmm, I'm not exactly sure how to make this work. When I use the suggested color for background, I get the grey color on hover, instead of just making the color of the button (which is blue) a bit lighter.

Reply:
You need to add a separate element that sits on top of the content. That's why I said `::after`, e.g. a pseudo-element that you create in CSS. I'd really recommend just moving to IconButton for this particular usage so you don't need to worry about this, and we can fix this in just the Button components instead of having to fix it all over the place.

Reply:
I did move this to the `IconButton` component, but I still need to have these styles for hover/active states. At least until the `IconButton` is made to do that automatically, for any color of the background it might have.

Reply:
Yeah, I was looking at this last night, I think we can build better implementations of these button classes that share things more cleanly now that hooks exist. The way things are setup atm feels pretty backwards (with content components, etc.). If you get this PR ready before I make that change you can merge this as-is and I will clean it up then